code/messaging/src/main/java/com/adobe/marketing/mobile/Proposition.java

```diff
@@ -34,9 +47,9 @@ public Proposition(final String uniqueId, final String scope, final Map<String,
         this.scope = scope;
         this.scopeDetails = scopeDetails;
         this.propositionItems = propositionItems;
-        for (final PropositionItem item: this.propositionItems) {
-            if (item.getProposition() == null) {
-                item.proposition = new WeakReference<>(this);
+        for (final PropositionItem item : this.propositionItems) {
+            if (item.getPropositionReference() == null) {
```

Comment:
We do not need to expose a getter for the reference as it's not a private property and accessible, we do need a getter for the referenced object though (as I mentioned)
```
if (item.propositionReference == null) {
    item.propositionReference = new SoftReference<>(this);
}
```